pkg/legacy/tarball/write.go

```diff
@@ -299,15 +249,50 @@ func MultiRefWrite(refToImage map[name.Reference]v1.Image, w io.Writer) error {
 	if err != nil {
 		return err
 	}
-	if err := tarball.WriteTarEntry(tf, "manifest.json", bytes.NewReader(tdBytes), int64(len(tdBytes))); err != nil {
+	if err := writeTarEntry(tf, "manifest.json", bytes.NewReader(tdBytes), int64(len(tdBytes))); err != nil {
 		return err
 	}
 	reposBytes, err := json.Marshal(&repos)
 	if err != nil {
 		return err
 	}
-	if err := tarball.WriteTarEntry(tf, "repositories", bytes.NewReader(reposBytes), int64(len(reposBytes))); err != nil {
+	if err := writeTarEntry(tf, "repositories", bytes.NewReader(reposBytes), int64(len(reposBytes))); err != nil {
 		return err
 	}
 	return nil
 }
+
+func dedupRefToImage(refToImage map[name.Reference]v1.Image) map[v1.Image][]string {
```

Comment:
You should be able to simplify this a lot now (no need to typecheck `name.Tag`).

Reply:
Why would that be the case? Can't caller still pass in references to `name.Digest` to `MultiWrite` or `Write` which will need to be handled here?